addons/website_sale/static/src/scss/website_sale.scss

```
@@ -42,6 +42,12 @@ $input-border-color: $gray-400;
     .o_wsale_context_thumb_2_3 {
         --o-wsale-card-thumb-aspect-ratio: 2/3;
     }
+    #o_wsale_products_grid.o_wsale_context_thumb_hidden {
+        .oe_product_image {
+            display: none;
+        }
+        grid-auto-rows: unset;
+    }
```

```suggestion
    #o_wsale_products_grid.o_wsale_context_thumb_hidden {
        grid-auto-rows: auto;

        .oe_product_image {
            display: none;
        }
    }
```

[Guideline](https://www.odoo.com/documentation/master/contributing/development/coding_guidelines.html#syntax-and-formatting) suggestion, properties & line break before nested selectors

You had doubt with the grid-auto-rows, from my testings it seems to work great and provides the condensed layout you're aiming for. Since unset makes it fallback to it's default `auto` value maybe we could explicitly set it to `auto` to ease readability 😄